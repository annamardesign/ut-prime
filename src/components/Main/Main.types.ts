import React from 'react';

export interface Props extends React.HTMLAttributes<HTMLDivElement> {
    routes: React.ReactNodeArray
}

export type StyledType = React.FC<Props>
