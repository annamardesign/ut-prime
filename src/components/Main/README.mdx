# Main

Internal component, no do not use it directly
