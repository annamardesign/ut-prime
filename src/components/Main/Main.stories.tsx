import React from 'react';
import { withReadme } from 'storybook-readme';

import Wrap from '../test/wrap';

// @ts-ignore: md file and not a module
import README from './README.md';
import Main from './index';

export default {
    title: 'Main',
    component: Main,
    decorators: [withReadme(README)]
};

const state = {
    preloadWindow: {
        toJS: () => ({open: false})
    },
    errorPopup: {
        open: false
    },
    login: {
        get: item => ({
            cookieChecked: true,
            authenticated: true,
            result: {
                getIn: () => {}
            }
        }[item])
    },
    portal: {
        tabs: [{
            title: 'Tab 1',
            path: '/tab1',
            component: () => function Component() { return <div>tab 1 body</div>; }
        }, {
            title: 'Tab 2',
            path: '/tab2',
            component: () => function Component() { return <div>tab 2 body</div>; }
        }]
    }
};

export const Basic: React.FC<{}> = () => <Wrap state={state}>
    <Main/>
</Wrap>;
