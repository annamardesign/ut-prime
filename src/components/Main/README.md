# Button

## How to use

```jsx
import Main from 'ut-front-devextreme/core/Main';

<Main onClick={handlerOnClick}>Hello Main</Main>;
```

## Props

- **className** - class applied to root `Main` html element.

| propName  | propType | defaultValue | isRequired |
| --------- | -------- | ------------ | ---------- |
| className | string   |              | no         |
