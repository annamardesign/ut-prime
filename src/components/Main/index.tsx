import React from 'react';
import {Route, Switch} from 'react-router';
import {hot} from 'react-hot-loader';

import Loader from '../Loader';
import Error from '../Error';
import Gate from '../Gate';
import Portal from '../Portal';

import { StyledType } from './Main.types';

const Main: StyledType = () => <Gate>
    <Switch>
        <Route
            component={
                () => <div style={{height: '100%'}}>
                    <Portal />
                    <Loader />
                    <Error />
                </div>
            }
        />
    </Switch>
</Gate>;

export default hot(module)(Main);
