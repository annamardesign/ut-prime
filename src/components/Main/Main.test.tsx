import React from 'react';

import { render} from '../test';
import Main from './index';

describe('<Main />', () => {
    it('render equals snapshot', () => {
        const { getByTestId } = render(
            <Main />,
            {},
            true
        );
        expect(getByTestId('ut-front-test')).toMatchSnapshot();
    });
});
