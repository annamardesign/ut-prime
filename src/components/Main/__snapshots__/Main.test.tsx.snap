// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> render equals snapshot 1`] = `
<div
  data-testid="ut-front-test"
>
  <div
    class="Gate-gate-1"
  />
</div>
`;
